src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -87,7 +85,7 @@ public RSAOpenSsl(SafeEvpPKeyHandle pkeyHandle)
         /// <returns>A SafeHandle for the RSA key in OpenSSL</returns>
         public SafeEvpPKeyHandle DuplicateKeyHandle()
         {
-            return Interop.Crypto.EvpPKeyDuplicate(GetKey(), Interop.Crypto.EvpAlgorithmId.RSA);
+            return GetKey().DuplicateHandle();
```

Comment:
We used to have a fail-safe that the algorithm was right.  Now we don't.  Why the change?

Reply:
IIRC EVP_PKEY_get0_RSA/EVP_PKEY_set1_RSA lost info about it being a provider key and didn't allow for any operations anymore after duplicating this way (at least for tpm2 provider). I'll add check in the constructor since I already added utility for key type